### PR TITLE
Amount input refactor for the PR '42 add confirmation page'

### PR DESCRIPTION
**Pull request by amonsosanz**
This PR implements a mechanism to recover the data of the user when the TX confirmation screen is closed, so the user can edit the wrong data and try again.

It also removes some wrong assumptions:

* Avoid assuming `0` as the default value for the default `TransactionData` in favor of `undefined`.
* Allow `undefined` as a valid value for the `amount-input` view onChange event.

### CODE REVIEW

**Comment by adrifdez** [comment]
We need to merge `env-refactor` because it's going to exist a conflict and your changes are in the conflict

File: src/views/home/components/amount-input/amount-input.view.tsx [hunk 1]
```diff
@@ -7,58 +7,65 @@ import Typography from "src/views/shared/typography/typography.view";
...
 
-  const updateAmountInput = (amount: BigNumber) => {
-    const newAmountWithFee = amount.add(actualFee);
-    const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(BigNumber.from(balance));
-    const isAmountInvalid = isNewAmountWithFeeMoreThanFunds || amount.isZero();
+const AmountInput: FC<AmountInputProps> = ({ value, token, balance, fee, onChange }) => {
+  const defaultInputValue = value ? getFixedTokenAmount(value.toString(), token.decimals) : "";
+  const [inputValue, setInputValue] = useState(defaultInputValue);
+  const classes = useAmountInputStyles(inputValue.length);
+  const actualFee = token.symbol === "WETH" ? fee : BigNumber.from(0);
```
Comment (elias-garcia) [inline src/views/home/components/amount-input/amount-input.view.tsx:31]: I've notice that this check is wrong, it should be: `token.address === ethers.constants.AddressZero`

Comment (amonsosanz) [reply src/views/home/components/amount-input/amount-input.view.tsx:31]: Good catch. I planned to add this comment to your env refactor PR, [where you removed some other hardcoded `weth`](https://github.com/hermeznetwork/bridge-ui/pull/45/files#diff-6f2b6981beaa48a10b9b87095264c4b66a18239e784739e99a2e97cacf1631a5L19).

File: src/views/home/components/transaction-form/transaction-form.view.tsx [hunk 2]
```diff
@@ -126,10 +138,10 @@ const TransactionForm: FC = () => {
         </div>
       </Card>
       <div className={classes.button}>
-        <Button type="submit" disabled={isInvalid}>
+        <Button type="submit" disabled={!localTransaction.amount || isInvalid}>
           Continue
         </Button>
-        {isInvalid && !localTransaction.amount.isZero() && <Error error="Insufficient balance" />}
+        {localTransaction.amount && isInvalid && <Error error="Insufficient balance" />}
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/views/home/components/transaction-form/transaction-form.view.tsx]: Fixed

**Review by elias-garcia — APPROVED** [review]

**Review by adrifdez — APPROVED** [review]

**Pull request merged**